src/util-logopenfile.c

```diff
@@ -516,6 +870,12 @@ int LogFileFreeCtx(LogFileCtx *lf_ctx)
             SCFree(lf_ctx->redis_setup.key);
     }
 #endif
+#ifdef HAVE_LIBRDKAFKA
+
+    if (lf_ctx->type == LOGFILE_TYPE_KAFKA) {
+        SCLogFileCloseKafka(lf_ctx);
```

Comment:
By the way the crash was due a bad termination sequence for rdkafka.

https://github.com/edenhill/librdkafka/wiki/Proper-termination-sequence